src/SvgStream.h

```diff
@@ -79,20 +80,31 @@ class SvgStreamString : public SvgStream {
   void write(char data)               { stream_ << data; }
   void write(const std::string& data) { stream_ << data; }
 
-  void seek(int n) {
-    stream_.seekg(n, std::ios_base::cur);
-  }
-
   void flush() {
-    stream_.flush();
-    env_["svg_string"] = stream_.str() + "</svg>";
   }
 
   void finish() {
-    flush();
-  }
+    // When device is closed, stream_ will be destroyed, so we can no longer
+    // get the svg string from stream_. In this case, we save the final string
+    // to the environment env, so that R can read from env$svg_string even
+    // after device is closed.
+    env_["is_closed"] = true;
```

Comment:
Or if you think it's really important that it should still return a value, then cache the value in the class. I think splitting the code across R and C++ is too confusing.